### PR TITLE
Fe feat/imageupload

### LINKED ISSUE #19: Component - ImageUploader

### PR DESCRIPTION
**Pull request by Elise-Jones**
# Description
-Create drag and drop functionality for user to add mushroom pictures
- Button for uploading a photo

Closes# (issue)
close #19 
## Type of change

Please delete options that are not relevant.

- [ x ] New feature (non-breaking change which adds functionality)

# Checklist:

- [ x ] My code follows the style guidelines of this project
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas
- [ x ] I have made corresponding changes to the documentation
- [ x ] My changes generate no new warnings 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Components/ImageUpload/ImageUpload.js [hunk 1]
```diff
@@ -59,16 +59,27 @@ const ImageUpload = () => {
...
-        <div  className="label-photo-input">
-          <p>Drag and Drop your mushroom find</p>
-            <button  className="upload-button" onClick={handleClick} >Upload Photo </button>
-        </div>
-        <p>{error}</p>
-      {userImage && <img src={userImage} />}
-      {/* above is only for viewing image droppings change state */}
-      {isDragging && <div className="drag-active" onDragEnter={handleDrag} onDragLeave={handleDrag} onDragOver={handleDrag} onDrop={handleDrop}></div> }
-    </form>
+    <input type="file" ref={userFileInput} id="photo-input" multiple={false} accept="image/*" onChange={handleFileUpload} />
```
Comment (jmartin777) [inline src/Components/ImageUpload/ImageUpload.js:64]: Great cleanup here!

File: src/Components/ImageUpload/ImageUpload.js [hunk 2]
```diff
@@ -59,16 +59,27 @@ const ImageUpload = () => {
...
+    {userImage ? (
```
Comment (jmartin777) [inline src/Components/ImageUpload/ImageUpload.js:65]: Great Job!

**Review by jmartin777 — APPROVED** [review]
Great start on this and taking the lead into it.!

File: src/Components/ImageUpload/ImageUpload.js [hunk 3]
```diff
@@ -1,8 +1,13 @@
-import React, { useState } from 'react';
+import React, { useState, useRef } from 'react';
 import "./ImageUpload.css"
 
 const ImageUpload = () => {
+  const [userImage, setUserImage] = useState(null)
   const [isDragging, setIsDragging] = useState(false);
+  const [base64Image, setBase64Image] = useState('')
+  const userFileInput = useRef(null)
+  const [imageError, setImageError] = useState(null)
```
Comment (andreasorensen) [inline src/Components/ImageUpload/ImageUpload.js:9]: Wow. So cool!

File: src/Components/ImageUpload/ImageUpload.js [hunk 4]
```diff
@@ -1,8 +1,13 @@
-import React, { useState } from 'react';
+import React, { useState, useRef } from 'react';
 import "./ImageUpload.css"
 
 const ImageUpload = () => {
+  const [userImage, setUserImage] = useState(null)
   const [isDragging, setIsDragging] = useState(false);
+  const [base64Image, setBase64Image] = useState('')
+  const userFileInput = useRef(null)
```
Comment (andreasorensen) [inline src/Components/ImageUpload/ImageUpload.js:8]: 🤩

File: src/Components/ImageUpload/ImageUpload.js [hunk 5]
```diff
@@ -14,39 +19,70 @@ const ImageUpload = () => {
...
-          <p>Drag and Drop your mushroom find</p>
-          <button className="upload-button" >
-            Upload Photo
-          </button>
-        </div>
-      </label>
-    </form>
+    <form id="photo-form" onDragEnter={handleDrag} onSubmit={(e) => e.preventDefault()}>
+    <input type="file" ref={userFileInput} id="photo-input" multiple={false} accept="image/*" onChange={handleFileUpload} />
+    {userImage ? (
```
Comment (andreasorensen) [inline src/Components/ImageUpload/ImageUpload.js:65]: Love the user of conditional rendering here.

File: src/Components/ImageUpload/ImageUpload.js (hunk 1 above)
Comment (andreasorensen) [reply src/Components/ImageUpload/ImageUpload.js:64]: Agreed. I love keeping that return statement clean!

**Review by andreasorensen — APPROVED** [review]
This is amazing. You crushed this Elise!

**Pull request merged**